src/main.ts

```diff
@@ -1211,6 +1236,39 @@ export default class RemotelySavePlugin extends Plugin {
     // Scans every 60 seconds
     this.vaultScannerIntervalId = window.setInterval(scanVault, 30_000);
   }
+
+  toggleStatusBarObserver(enabled: boolean) {
+    this.statusBarObserver?.disconnect();
+    this.statusBarObserver = undefined;
+
+    // Refresh status bar if new elements are found only if show only last synced is set.
+    if (enabled && this.settings.showLastSyncedOnly) {
+      this.statusBarObserver = new MutationObserver((mutationList, observer) => {
+        let shouldCall = false;
+        let byPlugin = false;
+
+        for (const mutation of mutationList) {
+          if (mutation.type === "childList" && mutation.addedNodes.length > 0) {
+            shouldCall = true;
+          }
+
+          mutation.addedNodes.forEach((node) => {
+            if ((node as Element).className === "status-bar-item plugin-remotely-secure") {
+              byPlugin = true;
```

Comment:
This is quite confusing to look at but, when the status bar element is added with toggleStatusBar it triggers the observer's callback. This is essentially just checking that the new element wasn't created by Remotely Sync. It also only calls toggleStatusBar if an element is added and ignores elements being removed.

Reply:
Appreciate you clarifying, thank you!